proto/src/determined/trial/v1/trial.proto

```diff
@@ -276,3 +276,33 @@ message MetricsReport {
   // Run ID of trial when metric was reported.
   int32 trial_run_id = 7;
 }
+
+// TrialSourceInfoType see \dT+ trial_source_info_type in db
+enum TrialSourceInfoType {
+  // The type is unspecified
+  TRIAL_SOURCE_INFO_TYPE_UNSPECIFIED = 0;
+  // "Inference" Trial Source Info Type, used for batch inference
+  TRIAL_SOURCE_INFO_TYPE_INFERENCE = 1;
+  // "Fine Tuning" Trial Source Info Type, used in model hub
+  TRIAL_SOURCE_INFO_TYPE_FINE_TUNING = 2;
+}
+
+// Denotes a connection between a given trial and a
+// checkpoint/source_trial/model_version
+message TrialSourceInfo {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: {
+      required: [ "trial_id", "checkpoint_uuid", "trial_source_info_type" ]
+    }
+  };
+  // ID of the trial.
+  int32 trial_id = 1;
+  // UUID of the checkpoint.
+  string checkpoint_uuid = 2;
+  // Source model_version id which generated the checkpoint (if applicable)
+  int32 source_model_version_id = 3;
+  // Source model_version version which generated the checkpoint (if applicable)
+  int32 source_model_version_version = 4;
```

Comment:
seems like these things are optional. can we:

```suggestion
  // Source model_version id which generated the checkpoint (if applicable)
  optional int32 source_model_version_id = 3;
  // Source model_version version which generated the checkpoint (if applicable)
  optional int32 source_model_version_version = 4;
```

2. Whenever we check if they are present, instead of checking they're 0, check for nil, i.e.

```
if tsi.SourceModelVersionId != nil {
```

Reply:
When I compile with `optional` I get this result:

```go
func (x *TrialSourceInfo) GetSourceModelVersionId() int32 {
	if x != nil && x.SourceModelVersionId != nil {
		return *x.SourceModelVersionId
	}
	return 0
}
```

Reply:
use the property `tsi.SourceModelVersionId` itself, not the getter method.